### PR TITLE
Add support for column specific fillna to viz

### LINKED ISSUE #3029: TypeError: unorderable types: str() > int() on specific distribution bar chart slice
Make sure these boxes are checked before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if any
- [x] I have reproduced the issue with at least the latest released version of superset
- [x] I have checked the issue tracker for the same issue and I haven't found one similar

### Superset version

0.18.5

### Expected results

the slice loads fine, as it did in caravel 0.14.X

### Actual results

```
2017-06-23 15:36:22,271:ERROR:root:unorderable types: str() > int()
Traceback (most recent call last):
  File "/venv/lib/python3.4/site-packages/superset/viz.py", line 240, in get_payload
    data = self.get_data(df)
  File "/venv/lib/python3.4/site-packages/superset/viz.py", line 1095, in get_data
    values=self.metrics)
  File "/venv/lib/python3.4/site-packages/pandas/core/reshape/pivot.py", line 160, in pivot_table
    table = table.sort_index(axis=1)
  File "/venv/lib/python3.4/site-packages/pandas/core/frame.py", line 3243, in sort_index
    labels = labels._sort_levels_monotonic()
  File "/venv/lib/python3.4/site-packages/pandas/core/indexes/multi.py", line 1240, in _sort_levels_monotonic
    indexer = lev.argsort()
  File "/venv/lib/python3.4/site-packages/pandas/core/indexes/base.py", line 2091, in argsort
    return result.argsort(*args, **kwargs)
TypeError: unorderable types: str() > int()
```

### Steps to reproduce

- open the slice

It happens on a specific slice that i cannot share unfortunately. Note to self: this is "distribuzione uso complessivo per scuola"

### PR DESCRIPTION
**Pull request by xrmx**
Fix #3029

### CODE REVIEW

File: tests/core_tests.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 from superset.models.sql_lab import Query
 from superset.views.core import DatabaseView
 from superset.connectors.sqla.models import SqlaTable
+from superset.viz import BaseViz
```
Comment (xrmx) [inline tests/core_tests.py]: not needed

**Comment by xrmx** [comment]
@mistercrunch any chance you can review this please?

**Comment by mistercrunch** [comment]
I'm not sure whether we want to systematically replace `NULL` with `<None>` and empties with `0`. I understand the use case but people may not want that in their existing charts....

**Comment by xrmx** [comment]
Maybe i wasn't clear but right now my visualization are not working with superset because NULL is not coerced to '0' (string) anymore as it was on caravel. So that fixes a regression for me :) Said that I don't think `<None>` is worst than `0` for text, we can change that to `<NULL>` so that would be more clear for SQL aware people. The surrounding `<>` are just a trick to have it show before other letters after sorting. For other types the 0 is actually hardcoded so I am not changing any behaviour.

**Comment by xrmx** [comment]
@mistercrunch any hint on how we can move this forward?

**Comment by ToonoW** [comment]
good job

**Comment by ToonoW** [comment]
It make sense what @mistercrunch say. We should not use 0 and ‘NULL’ replace `<null>` in sample way, it will make misunderstanding in some situation.

Simple and Difficult Problem.  : (

**Comment by rumbin** [comment]
Just wanted you to know that I would also love to see #3029 fixed in the next release, as some of our pivot tables also don't work any longer when upgrading from 0.15.4 to 0.19.1.

So, thanks @xrmx for your efforts!

**Pull request merged**